### ISSUE TITLE
Bug: Lfortran 0.36.0 was right, 0.40.0 is wrong with list-directed output of array section

### ISSUE THREAD

**Issue by harperjf** [lfortran/lfortran#4705]
Lfortran 0.40.0 in my system (now Ubuntu 24.04) failed to insert the blank space between 32 and 2147483647 which was required for * (list-directed) output between two integers , which were an array section, in my program arraysectiond.f90. The output was on two lines but should have been on one. Lfortran 0.36.0 was correct in both respects (see #4501, which also reported another problem). The program:
```
(lf) john:~/Test$ cat arraysectiond.f90
  implicit none
  integer,parameter:: k = selected_int_kind(9) ! default integer
  integer :: iarray(3,4)
  iarray(1,1:2) = [bit_size(1_k),huge(1_k)]
  print *,kind(1_k),iarray(1,1:2)
end program
```
Output from lfortran 0.40.0 (wrong) :
```
4
322147483647
```
Output from lfortran 0.36.0 (right) :
```
4 32 2147483647
```

**Comment by kmr-srbh** [comment]
Thanks for reporting this! We recently made improvements to the way we handle `print`. This is a bug we need to fix. @assem2002

**Comment by assem2002** [comment]
Yeah, that's because of how `print_arr` pass handles array printing when with stringformat.
before print refatorings we didn't use stringformat, while now we always use stringformat. and printing arrays with stringformat leads to this weird behavior.

I'm intending to limit the work done on print by `print_arr` pass, it should be fixed.
does that need to be solved immediately?

**Comment by HarshitaKalani** [comment]
It works now. Let's add a test to close this.
```console
% lfortran a.f90 
4    32    2147483647
% gfortran a.f90 && ./a.out
           4          32  2147483647
```
